From 24af3fa28d78ed07311f05f87e71dc9b5bbef757 Mon Sep 17 00:00:00 2001
From: Tom Zanussi <tom.zanussi@linux.intel.com>
Date: Wed, 28 Aug 2013 14:24:09 -0500
Subject: [PATCH 004/429] Revert "timer_list: Split
 timer_list_show_tickdevices"

This reverts commit 60cf7ea849e77c8782dee147cfb8c38d1984236e.

This is a temporary workaround for the dropbearkey hang issue in 3.10.
---
 kernel/time/timer_list.c | 21 ++++++++++++---------
 1 file changed, 12 insertions(+), 9 deletions(-)

diff --git a/kernel/time/timer_list.c b/kernel/time/timer_list.c
index 7d025cc41f6a..819d3493ca09 100644
--- a/kernel/time/timer_list.c
+++ b/kernel/time/timer_list.c
@@ -133,6 +133,7 @@ static void print_cpu(struct seq_file *m, int cpu, u64 now)
 	struct hrtimer_cpu_base *cpu_base = &per_cpu(hrtimer_bases, cpu);
 	int i;
 
+	SEQ_printf(m, "\n");
 	SEQ_printf(m, "cpu: %d\n", cpu);
 	for (i = 0; i < HRTIMER_MAX_CLOCK_BASES; i++) {
 		SEQ_printf(m, " clock %d:\n", i);
@@ -186,7 +187,6 @@ static void print_cpu(struct seq_file *m, int cpu, u64 now)
 
 #undef P
 #undef P_ns
-	SEQ_printf(m, "\n");
 }
 
 #ifdef CONFIG_GENERIC_CLOCKEVENTS
@@ -195,6 +195,7 @@ print_tickdevice(struct seq_file *m, struct tick_device *td, int cpu)
 {
 	struct clock_event_device *dev = td->evtdev;
 
+	SEQ_printf(m, "\n");
 	SEQ_printf(m, "Tick Device: mode:     %d\n", td->mode);
 	if (cpu < 0)
 		SEQ_printf(m, "Broadcast device\n");
@@ -229,11 +230,12 @@ print_tickdevice(struct seq_file *m, struct tick_device *td, int cpu)
 	print_name_offset(m, dev->event_handler);
 	SEQ_printf(m, "\n");
 	SEQ_printf(m, " retries:        %lu\n", dev->retries);
-	SEQ_printf(m, "\n");
 }
 
-static void timer_list_show_tickdevices_header(struct seq_file *m)
+static void timer_list_show_tickdevices(struct seq_file *m)
 {
+	int cpu;
+
 #ifdef CONFIG_GENERIC_CLOCKEVENTS_BROADCAST
 	print_tickdevice(m, tick_get_broadcast_device(), -1);
 	SEQ_printf(m, "tick_broadcast_mask: %08lx\n",
@@ -244,7 +246,12 @@ static void timer_list_show_tickdevices_header(struct seq_file *m)
 #endif
 	SEQ_printf(m, "\n");
 #endif
+	for_each_online_cpu(cpu)
+		print_tickdevice(m, tick_get_device(cpu), cpu);
+	SEQ_printf(m, "\n");
 }
+#else
+static void timer_list_show_tickdevices(struct seq_file *m) { }
 #endif
 
 static int timer_list_show(struct seq_file *m, void *v)
@@ -272,16 +279,12 @@ void sysrq_timer_list_show(void)
 	SEQ_printf(m, "Timer List Version: v0.7\n");
 	SEQ_printf(m, "HRTIMER_MAX_CLOCK_BASES: %d\n", HRTIMER_MAX_CLOCK_BASES);
 	SEQ_printf(m, "now at %Ld nsecs\n", (unsigned long long)now);
-	SEQ_printf(m, "\n");
 
 	for_each_online_cpu(cpu)
 		print_cpu(m, cpu, now);
 
-#ifdef CONFIG_GENERIC_CLOCKEVENTS
-	timer_list_show_tickdevices_header(m);
-	for_each_online_cpu(cpu)
-		print_tickdevice(m, tick_get_device(cpu), cpu);
-#endif
+	SEQ_printf(m, "\n");
+	timer_list_show_tickdevices(m);
 
 static void *move_iter(struct timer_list_iter *iter, loff_t offset)
 {
-- 
2.37.3

